packages/dotcom-ui-app-context/src/client/AppContext.ts

```diff
@@ -1,20 +1,12 @@
+import { ClientEmbedData } from '@financial-times/dotcom-ui-client-embed/src/client'
 import { TAppContext } from '../types'
 
 export type TAppContextOptions = {
   appContext: TAppContext
 }
-export default class AppContext {
-  private appContext: TAppContext
 
+export default class AppContext extends ClientEmbedData {
   constructor(options: TAppContextOptions) {
-    this.appContext = Object.freeze(options.appContext)
-  }
-
-  get(property: string): any {
-    return this.appContext.hasOwnProperty(property) ? this.appContext[property] : undefined
-  }
-
-  getAll(): TAppContext {
-    return this.appContext
+    super(options.appContext)
```

Comment:
Need to figure out how to extend a class and apply the types, as this currently has removed the type information.

Reply:
Looks like what I wanted to do is not possible(?) so will need to figure out another way to achieve this: https://github.com/microsoft/TypeScript/issues/3402